src/tcpclientsrc.rs

```diff
@@ -0,0 +1,726 @@
+// Copyright (C) 2018 Sebastian Dröge <sebastian@centricular.com>
+// Copyright (C) 2018 LEE Dongjun <redongjun@gmail.com>
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Library General Public
+// License as published by the Free Software Foundation; either
+// version 2 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Library General Public License for more details.
+//
+// You should have received a copy of the GNU Library General Public
+// License along with this library; if not, write to the
+// Free Software Foundation, Inc., 51 Franklin Street, Suite 500,
+// Boston, MA 02110-1335, USA.
+
+use glib;
+use glib::prelude::*;
+use gst;
+use gst::prelude::*;
+
+use gobject_subclass::object::*;
+use gst_plugin::element::*;
+
+use std::sync::Mutex;
+use std::u16;
+
+use futures;
+use futures::future;
+use futures::Future;
+use tokio::net;
+
+use either::Either;
+
+use rand;
+
+use iocontext::*;
+use socket::*;
+
+const DEFAULT_ADDRESS: Option<&'static str> = Some("127.0.0.1");
+const DEFAULT_PORT: u32 = 5000;
+const DEFAULT_CAPS: Option<gst::Caps> = None;
+const DEFAULT_MTU: u32 = 1500;
+const DEFAULT_CONTEXT: &'static str = "";
+const DEFAULT_CONTEXT_THREADS: i32 = 0;
+const DEFAULT_CONTEXT_WAIT: u32 = 0;
+
+#[derive(Debug, Clone)]
+struct Settings {
+    address: Option<String>,
+    port: u32,
+    caps: Option<gst::Caps>,
+    mtu: u32,
+    context: String,
+    context_threads: i32,
+    context_wait: u32,
+}
+
+impl Default for Settings {
+    fn default() -> Self {
+        Settings {
+            address: DEFAULT_ADDRESS.map(Into::into),
+            port: DEFAULT_PORT,
+            caps: DEFAULT_CAPS,
+            mtu: DEFAULT_MTU,
+            context: DEFAULT_CONTEXT.into(),
+            context_threads: DEFAULT_CONTEXT_THREADS,
+            context_wait: DEFAULT_CONTEXT_WAIT,
+        }
+    }
+}
+
+static PROPERTIES: [Property; 7] = [
+    Property::String(
+        "address",
+        "Address",
+        "Address to receive packets from",
+        DEFAULT_ADDRESS,
+        PropertyMutability::ReadWrite,
+    ),
+    Property::UInt(
+        "port",
+        "Port",
+        "Port to receive packets from",
+        (0, u16::MAX as u32),
+        DEFAULT_PORT,
+        PropertyMutability::ReadWrite,
+    ),
+    Property::Boxed(
+        "caps",
+        "Caps",
+        "Caps to use",
+        gst::Caps::static_type,
+        PropertyMutability::ReadWrite,
+    ),
+    Property::UInt(
+        "mtu",
```

Comment:
TCP being a stream, MTU makes no sense here. I'd call this chunk size for example